### DISCUSSION TITLE
What should I do if I want to use Arduino to compile?

### DISCUSSION THREAD

**Discussion by zhouxule** [grillbaer/esp32-geiger-counter#6] [Q&A]
Hi,
I only have VS (not VS code) and Arduino on my PC. How can I use Arduino to compile the program? 

Thanks and best regards. :)

**Comment by grillbaer** [comment]
Hi @zhouxule,

I recommend using VS Code and its PlatformIO plug-in. VS Code can be installed in parallel to Visual Studio without any problems, it is completely free and many developers really like it. Note that Visual Studio and VS Code are two completely independent IDEs with somehow similar names.

If you still want to try it with the Arduino IDE, I guess you need to install its ESP32 add-on and the libraries listed in this project's file `platformio.ini` (U8g2, 256dpi) by hand and copy the source files from the project's `src` folder to the IDE. Maybe it also complains about some `#include` statements in the project source files and you have to remove them or find the right way of handling. Since I have no real experience with the Arduino IDE, this is only guessing.

Best regards,
Holger
